mapie/calibration.py

```diff
@@ -0,0 +1,502 @@
+                random_state=random_state,
+                shuffle=shuffle,
+                stratify=stratify,
+            )
+            X_train, y_train, X_calib, y_calib, sw_train, sw_calib = results
+            X_train, y_train = indexable(X_train, y_train)
+            y_train = _check_y(y_train)
+            sw_train, X_train, y_train = check_null_weight(
+                sw_train,
+                X_train,
+                y_train
+            )
+            self.n_classes_ = len(np.unique(y_train))
+            self.estimator = fit_estimator(
+                clone(estimator), X_train, y_train, sw_train,
+            )
+            self.classes_ = self.estimator.classes_  # type: ignore
+            self.calibrators = self._fit_calibrators(
+                X_calib, y_calib, sw_calib, self.estimator, calibrator
+            )
+        return self
+
+    def predict_proba(
+        self,
+        X: ArrayLike,
+    ) -> NDArray:
+        """
+        Prediction of the calibrated probability score of the class after
+        fitting of the classifer and calibrator.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Test data.
+
+        Returns
+        -------
+        NDArray of shape (n_samples, n_classes)
+            The calibrated score for each max score and zeros at every
+            other position in that line.
+        """
+        check_is_fitted(self, self.fit_attributes)
+        y_pred_probs = self.estimator.predict_proba(X=X)  # type: ignore
+        # y_pred_probs = fix_number_of_classes(
+        #     self.n_classes_,
+        #     self.estimator.classes_,  # type: ignore
+        #     y_pred_probs
+        # )
+        self.uncalib_pred = y_pred_probs
+
+        max_prob, max_prob_arg = self._get_labels(
+            cast(str, self.method),
+            y_pred_probs
+        )
+
+        n = _num_samples(max_prob)
+        calibrated_test_values = np.zeros((n, self.n_classes_))
+
+        for idx, item in enumerate(np.unique(max_prob_arg)):
+            calibrated_test_values = self._pred_proba_calib(
+                idx,
+                item,
+                calibrated_test_values,
+                max_prob,
+                max_prob_arg,
+                self.calibrators,
+            )
+        return calibrated_test_values
+
+    def predict(
+        self,
+        X: ArrayLike,
+    ) -> NDArray:
+        """
+        Predict the class of the estimator after calibration.
+        Note that in the top-label setting, this class does not change.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Test data.
+
+        Returns
+        -------
+        NDArray of shape (n_samples,)
+            The class from the predictions.
+        """
+        check_is_fitted(self, self.fit_attributes)
+        return self.classes_[
+            np.argmax(self.predict_proba(X=X), axis=1)
+        ]
```

Comment:
use model.predict directly

Reply:
I believe that this way is more general and better reflects what should occur in calibration. For top-label we don't change the label, but it would be possible that with another method we do. I believe this better reflects the theoretical output.

Reply:
with this line of code, you will have to change your code is a new method change the predicted label. I think, it just makes your code heavier and less readable

Reply:
okay, done.